File: app/src/main/java/vn/kingbee/widget/dialog/ext/help/HelpMode.java
package vn.kingbee.widget.dialog.ext.help;

public enum HelpMode {
    DEFAULT_SCREEN("default_screen"),
    HOME_SCREEN("home_screen"),
    LOGIN_SCREEN("login_screen"),
    LOGIN_ID_CELLPHONE_SCREEN("login_id_cellphone_screen"),
    LOGIN_FINGER_SCREEN("login_finger_screen");
    private final String mName;

    HelpMode(String name) {
        mName = name;
    }

    public static HelpMode getScreenFromName(String name) {
        for (HelpMode c : HelpMode.values()) {
            if (c.getName().equalsIgnoreCase(name))
                return c;
        }
        return DEFAULT_SCREEN;
    }

    public String getName() {
        return mName;
    }
}

File: app/src/main/java/vn/kingbee/widget/dialog/ext/help/HelpMode.kt
package vn.kingbee.widget.dialog.ext.help

enum class HelpMode private constructor(name: String) {
    DEFAULT_SCREEN("default_screen"),
    HOME_SCREEN("home_screen"),
    LOGIN_SCREEN("login_screen"),
    LOGIN_ID_CELLPHONE_SCREEN("login_id_cellphone_screen"),
    LOGIN_FINGER_SCREEN("login_finger_screen");


    companion object {

        fun getScreenFromName(name: String): HelpMode {
            for (c in HelpMode.values()) {
                if (c.name.equals(name, ignoreCase = true))
                    return c
            }
            return DEFAULT_SCREEN
        }
    }
}